### DISCUSSION TITLE
Adding cms page section to admin area. #NextJS #Vendure

### DISCUSSION THREAD

**Discussion by Skatedan26** [vendurehq/vendure#1061] [Q&A]
Hi, I'm doing some tests on vendure as administration and next.js as frontend, I'm new to both areas.

I would need to understand how I could add the possibility to create / modify / delete some cms pages in the administrator panel, could you kindly give me support?

thanks a lot

Dan

**Comment by michaelbromley** [comment]
Hi Dan,

I think this depends on the CMS you plan to use. In the simplest case, you could create a [ui extension](https://www.vendure.io/docs/plugins/extending-the-admin-ui/) which iframes the CMS admin app. However, then you'd need to log in twice and it's probably no better than just using the CMS admin app in a separate tab.

If the CMS is headless, you could create a ui extension which uses its API to add CRUD functionality on the CMS directly from the Vendure admin ui. I'd imagine that this would be quite a lot of work though to get right.

How do you imagine it working in the ideal scenario?

**Comment by Skatedan26** [comment]
Hi @michaelbromley, thank you for quick and response to my question! :)
I would like to manage cms from the Verdure user interface, which once the cms page has been created, its routing will be handled by nextJS.

I think that it is necessary store content of my custom page on DB, I'd work on GraphiQL, right?
There are some example that i could follow ?

**Comment by michaelbromley** [comment]
Ah so if you want to implement your own simple CMS pages, then yes you can create a plugin which adds e.g. a "page" entity to the DB and exposes it via the GraphQL API. Please take a look at the examples given here for pointers: https://www.vendure.io/docs/plugins/plugin-examples/

**Comment by Skatedan26** [comment]
what would be the step by step logic I should follow to do a correct job?

**Reply by michaelbromley** [reply]
Here's a step-by-step guide to create a plugin: https://www.vendure.io/docs/plugins/writing-a-vendure-plugin/

Here's a guide to extend the admin ui once you implemented your plugin: https://www.vendure.io/docs/plugins/extending-the-admin-ui/using-angular/

**Reply by Skatedan26** [reply]
ok, I'll take a look at it, thank you!!
